Enable ITS encryption by default on 54L + other TF-M updates (#15418)
nrfconnect/sdk-nrf: samples/crypto/persistent_key_usage/README.rst

The persistent key sample shows how to generate a persistent key using the Platform Security Architecture (PSA) APIs.
Persistent keys are stored in the Internal Trusted Storage (ITS) of the device and retain their value between resets.

> The ITS backend is either provided by TF-M or the :ref:`trusted_storage_readme` library for non TF-M builds.

Reviewer: ```suggestion
The ITS backend is either provided by TF-M, or the :ref:`trusted_storage_readme` library when building applications without TF-M.
```

   The key ID of a read-only key is writable again after a full erase of the device memory.
   Use the ``west -v flash --erase`` command for the full erase.

.. note::

>    For non-TF-M builds, and on nRF54L15 devices, the :ref:`hardware unique key (HUK) <lib_hw_unique_key>` is used to encrypt the key before storing it.`

Reviewer: ```suggestion
   In environments without TF-M and on nRF54L15 devices, the :ref:`hardware unique key (HUK) <lib_hw_unique_key>` is used to encrypt the key before storing it.`
```

Author: Discussed offline an changed

1. |connect_terminal|
#. Compile and program the application.
#. Observe the logs from the application using a terminal emulator.

Dependencies
************

* PSA APIs:

   * :file:`psa/crypto.h`

> * Non TF-M builds use the :ref:`trusted_storage_readme` library

Reviewer: ```suggestion
* Builds without TF-M use the :ref:`trusted_storage_readme` library
```
